Replace jquery-ui blind and fade effects with jquery slide and fade effects

martasiewierska opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description (*)
<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->
Jquery UI effects seem to be rarely used. I was able to find only 2 places - minicart quantity update button and some checkout error messages. Unfortunately as minicart and authentication popup are loaded on every page quite a big chunk of code is loaded. It seems to me that these places come from quite old times and can be replaced with jquery animations - slideUp of fadeIn/Out. I cannot see any difference in behavior after replacing them. The advantage is that jquery-ui effects are not loaded at all which can improve performance.

### Related Pull Requests
<!-- related pull request placeholder -->

### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/magento2#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.

1. Fixes magento/magento2#<issue_number>-->

### Manual testing scenarios (*)
<!---
    Please provide a set of unambiguous steps to test the proposed code change.
    Giving us manual testing scenarios will help with the processing and validation process.
-->
1. Add any product to the cart. Open minicart. Change the quantity of the product. Check if the update button appears with fade effect.
2. Click "Update" and check if minicart quantity update button disappears wit fade effect
3. Disable guest checkout. Add any product to the cart and click "Go to checkout". provide wrong credentials. An error message should appear.
4. Wait for 15s. After this time error message should disappear with slide effect.
5. Got to payment step of onepage checkout. Provide wrong data for any payment method (like missing credit card number) and click the "Place order" button
6. Error message should appear. Wait for 15s. After this time error message should disappear with slide effect.

### Questions or comments
<!---
	If relevant, here you can ask questions or provide comments on your pull request for the reviewer
	For example if you need assistance with writing tests or would like some feedback on one of your development ideas
-->

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [x] All automated tests passed successfully (all builds are green)

msiewierska committed:
Replace jquery-ui blind and fade effects with jquery slide and fade effects

msiewierska committed:
Fix hideSpeed time changed by mistake

martasiewierska commented:
@magento run all tests

mrtuvn commented:
Thank you for such another great improvements. Let's see final tests results

ihor-sviziev commented:
@magento run Functional Tests EE

Den4ik reviewed (commented):
Hi @martasiewierska Thanks for you contribution. But I'm not sure about
> big chunk of code

Described in this PR ui modules included as separate files
https://github.com/magento/magento2/blob/2.4-develop/lib/web/jquery/ui-modules/effect-blind.js (2.35KB) and https://github.com/magento/magento2/blob/2.4-develop/lib/web/jquery/ui-modules/effect-fade.js (660Bytes)
Unfortunately end user will not save huge traffic and see performance effect.

ihor-sviziev commented:
@Den4ik anyway, fixing all such things a bit improving page load time in general. We should reduce loaded JS files continuously, even while removing specific js files from the load isn't a huge change.
So I'm for accepting this PR.

martasiewierska commented:
Yes, the file size rather small. However, I saw a small improvement in Lighthouse rating after applying this change as a patch (+ also changes from my other PRs - removing draggable, resizable jquery-ui chunks). Clients are really pushing toward better performance and reducing unused js size is one of the opportunities.

martasiewierska commented:
Yes, the file size rather small. However, I saw a small improvement in Lighthouse rating after applying this change as a patch (+ also changes from my other PRs - removing draggable, resizable jquery-ui chunks). Clients are really pushing toward better performance and reducing unused js size is one of the opportunities.

martasiewierska commented:
I closed this PR by mistake and then reopened it. I am sorry.

Den4ik commented:
@ihor-sviziev I totally agree that we should reduce count if dependancy.
My point is that unfortunately performance increase is not so high as wanted. Also you will not see any performance increase if use of js merging and minification (as expected on production).
But I'm agree with this changes and thanks @martasiewierska that you a looking ways for increasing Magento preformance.

**P.S.** I don't like jq effects like fadeIn because I faced with very slow execution and high CPU load on items count > 30. So I vote for standard show/hide without any effects :)

magento-engcom-team commented:
Hi @Den4ik, thank you for the review.
[ENGCOM-9054](https://jira.corp.magento.com/browse/ENGCOM-9054) has been created to process this Pull Request

ihor-sviziev commented:
That's actually a good point that such animations might cause CPU load. I think the optimal way would be to move the logic to CSS classes with css transitions.
@Den4ik @martasiewierska what do you think about that?

Den4ik commented:
> That's actually a good point that such animations might cause CPU load. I think the optimal way would be to move the logic to CSS classes with css animation, like this https://stackoverflow.com/questions/11679567/using-css-for-a-fade-in-effect-on-page-load.
> @Den4ik @martasiewierska what do you think about that?

Yep, css effects preferred for me by their performance

ihor-sviziev commented:
@martasiewierska will you be able to update your solution to css based?

martasiewierska commented:
@Den4ik I am sorry for such a long response. Yes, I had in mind css animations and I could introduce them. However, I am not sure if there is any convention for active classes. I checked briefly and it seems sometimes there is `_active` or `active` class, but also `show`, `_show` but also `shown`. It seems to me that till now css and corresponding classes were not used in Magento for transitions, I see that rather border colors or other small things are transitioned. What class should I use in your opinion on the active state? Is there any convention?

Den4ik commented:
@martasiewierska I could not especially recommend exists classes. In my opinion `show` is more correct, but it could affect multiple places

msiewierska committed:
Change jquery fade into css opacity animation for update qty minicart button

martasiewierska commented:
I removed fadeIn/Out effect for minicart qty update button and replaced it with css transition.
It seems to me that introducing css transition effect for error messages in message.js file will not be possible. We can only identify elements to be slided by `selector: '[data-role=checkout-messages]',` As we do not have css class, only data-role, it won't be possible to add transition styles.

ihor-sviziev commented:
@magento run all tests

ihor-sviziev commented:
Looks good to me. Let's wait for test results

ihor-sviziev reviewed (changes requested):
Could you fix failing static test?

Den4ik commented:
@martasiewierska Could you fix failing static test?

mrtuvn commented on app/design/frontend/Magento/blank/Magento_Checkout/web/css/source/module/_minicart.less:
```diff
@@ -354,6 +355,15 @@
+            }
         }
 
         .subtitle {
```
Can you re-sort order of properites following alphabetically

Den4ik reviewed (changes requested):
@martasiewierska Could you please fix static test

ihor-sviziev commented:
Hi @martasiewierska,
Will you be able to fix failing static tests?

ihor-sviziev commented on app/design/frontend/Magento/blank/Magento_Checkout/web/css/source/module/_minicart.less:
```diff
@@ -354,6 +355,15 @@
             .lib-font-size(11);
             margin-left: 5px;
             vertical-align: top;
+            pointer-events: none;
+            opacity: 0;
+            .lib-css(transition, @minicart-update-cart-item__transition);
```
```suggestion
            .lib-font-size(11);
            .lib-css(transition, @minicart-update-cart-item__transition);
            margin-left: 5px;
            opacity: 0;
            pointer-events: none;
            vertical-align: top;
```

ihor-sviziev committed:
Replace jquery-ui blind and fade effects with jquery slide and fade effects

Fix static tests

ihor-sviziev commented:
@magento run Static Tests

ihor-sviziev commented:
@magento run all tests

mrtuvn commented:
Is current maintainers can able to modify/push to author branch ?

mrtuvn commented on app/code/Magento/Checkout/view/frontend/web/template/minicart/item/default.html:
```diff
@@ -88,8 +88,7 @@
-                            class="update-cart-item"
-                            style="display: none">
+                            class="update-cart-item">
                         <span data-bind="i18n: 'Update'"></span>
```
Now button update in minicart always show by default ?

ihor-sviziev replied:
yeah... feels like it just having opacity 0...

mrtuvn replied:
Can we update for luma theme also ? opacity only apply for blank theme

leeroybrun replied:
> Hi @martasiewierska Thanks for you contribution. But I'm not sure about
> 
> > big chunk of code
> 
> Described in this PR ui modules included as separate files
> https://github.com/magento/magento2/blob/2.4-develop/lib/web/jquery/ui-modules/effect-blind.js (2.35KB) and https://github.com/magento/magento2/blob/2.4-develop/lib/web/jquery/ui-modules/effect-fade.js (660Bytes)
> Unfortunately end user will not save huge traffic and see performance effect.

Please note that these files also seems to load the file https://github.com/magento/magento2/blob/2.4-develop/lib/web/jquery/ui-modules/effect.js (46.7 KB).

With the changes in this PR, this file will not be loaded anymore, and so we will save around 50 KB total.

andrewbess commented:
Hello @ihor-sviziev 
The internal team believes that this pull request is not related to the platform's health project.
Also, this pull request based on 2.4-develop.
Sorry.

leeroybrun commented:
> Hi @martasiewierska Thanks for you contribution. But I'm not sure about
> 
> > big chunk of code
> 
> Described in this PR ui modules included as separate files
> https://github.com/magento/magento2/blob/2.4-develop/lib/web/jquery/ui-modules/effect-blind.js (2.35KB) and https://github.com/magento/magento2/blob/2.4-develop/lib/web/jquery/ui-modules/effect-fade.js (660Bytes)
> Unfortunately end user will not save huge traffic and see performance effect.

Please note that these files also seems to load the file https://github.com/magento/magento2/blob/2.4-develop/lib/web/jquery/ui-modules/effect.js (46.7 KB).

With the changes in this PR, this file will not be loaded anymore, and so we will save around 50 KB total.

magento-engcom-team commented:
Hi @Den4ik, thank you for the review.
[ENGCOM-9054](https://jira.corp.magento.com/browse/ENGCOM-9054) has been created to process this Pull Request

mrtuvn commented:
`Can we update for luma theme also ? opacity only apply for blank theme`
Can you update for this @martasiewierska

leonhelmus commented:
Can we merge this PR?

mrtuvn commented:
i think this efforts deserves give more attention
cc: @sidolov @sdzhepa

engcom-Dash commented:
Hello @martasiewierska ,

Thank you for your contributions!

Please resolve the conflict so we can proceed with this PR.

Thanks again!

engcom-Dash commented:
Closing this PR since it has not been updated in a while. Please feel free to reopen if needed.
